src/hssm/link.py

```diff
@@ -0,0 +1,78 @@
+"""A class that extends bmb.Link to allow for more generalized links with bounds."""
+
+import bambi as bmb
+import numpy as np
+
+HSSM_LINKS = {"gen_logit"}
+
+
+class Link(bmb.Link):
+    """Representation of a generalized link function.
+
+    This object contains two main functions. One is the link function itself, the
+    function that maps values in the response scale to the linear predictor, and the
+    other is the inverse of the link function, that maps values of the linear predictor
+    to the response scale.
+
+    The great majority of users will never interact with this class unless they want to
+    create a custom ``Family`` with a custom ``Link``. This is automatically handled for
+    all the built-in families.
+
+    Parameters
+    ----------
+    name : str
+        The name of the link function. If it is a known name, it's not necessary to pass
+        any other arguments because functions are already defined internally. If not
+        known, all of `link``, ``linkinv`` and ``linkinv_backend`` must be specified.
+    link : function
+        A function that maps the response to the linear predictor. Known as the
+        :math:`g` function in GLM jargon. Does not need to be specified when ``name``
+        is a known name.
+    linkinv : function
+        A function that maps the linear predictor to the response. Known as the
+        :math:`g^{-1}` function in GLM jargon. Does not need to be specified when
+        ``name`` is a known name.
+    linkinv_backend : function
+        Same than ``linkinv`` but must be something that works with PyMC backend
+        (i.e. it must work with PyTensor tensors). Does not need to be specified when
+        ``name`` is a known name.
+    """
+
+    def __init__(
+        self,
+        name,
+        link=None,
+        linkinv=None,
+        linkinv_backend=None,
+        bounds: tuple[float, float] | None = None,
+    ):
+        if name in HSSM_LINKS:
+            self.name = name
+            if name == "gen_logit":
+                if bounds is None:
+                    raise ValueError(
+                        "Bounds must be specified for generalized log link function."
+                    )
+                self.link = self._make_generalized_logit_simple(*bounds)
+                self.linkinv = self._make_generalized_sigmoid_simple(*bounds)
+                self.linkinv_backend = self._make_generalized_sigmoid_simple(*bounds)
+        else:
+            bmb.Link.__init__(name, link, linkinv, linkinv_backend)
+
+        self.bounds = bounds
+
+    def _make_generalized_sigmoid_simple(self, a, b):
```

Comment:
I think you want to use pytensor operations here no?
At least the linkinv_backend should be pytensor, the others are prob ok.
The example I gave had all pytensor? I think so. (So all pytensor should work either way, since I tested that)

Reply:
Actually `bambi` uses `numpy` functions, so I just want to make it consistent with other link functions in bambi. https://github.com/bambinos/bambi/blob/main/bambi/families/link.py

Reply:
kind of confusing because in the docs for Link it says:
'''
    linkinv_backend : function
        Same than ``linkinv`` but must be something that works with PyMC backend (i.e. it must
        work with PyTensor tensors). Does not need to be specified when ``name`` is a known
        name.
        '''

Reply:
do numpy --> pt happen automatically?
because without pt it's not going to be differentiable...

Reply:
I wouldn't worry too much about this. I basically copied the docs from Bambi, and numpy should be compatible with pytensor. Also, we are not touching `linkinv_backend` here. Actually none of the bambi-defined link functions uses `link_backend`

Reply:
> do numpy --> pt happen automatically? because without pt it's not going to be differentiable...

I think there's some interpretability between numpy and pytensor. Otherwise none of the bambi link functions would work

Reply:
ok!